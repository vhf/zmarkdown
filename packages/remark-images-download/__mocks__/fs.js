const fs = jest.genMockFromModule('fs')

fs.stat = (path, callback) => {
  const err = null
  const stats = {
    isDirectory () {
      return true
    }
  }
  process.nextTick(() => callback(err, stats))
}

fs.mkdir = (path, callback) => {
  process.nextTick(() => callback())
}

module.exports = fs
