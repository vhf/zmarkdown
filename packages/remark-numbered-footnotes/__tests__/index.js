import {readdirSync as directory, readFileSync as file} from 'fs'
import {join} from 'path'
import ava from 'ava'
import plugin from '..'
import unified from 'unified'
import reParse from 'remark-parse'
import stringify from 'rehype-stringify'
import remark2rehype from 'remark-rehype'

const base = join(__dirname, 'fixtures')

const specs = directory(base).reduce((tests, contents) => {
  const parts = contents.split('.')
  if (!tests[parts[0]]) {
    tests[parts[0]] = {}
  }
  tests[parts[0]][parts[1]] = file(join(base, contents), 'utf-8')
  return tests
}, {})

Object.keys(specs).filter(Boolean).forEach(name => {
  const spec = specs[name]

  ava(name, t => {
    const {contents} = unified()
      .use(reParse, {
        footnotes: true
      })
      .use(plugin)
      .use(remark2rehype)
      .use(stringify)
      .processSync(spec.fixture)

    t.deepEqual(contents, spec.expected.trim())
  })
})
