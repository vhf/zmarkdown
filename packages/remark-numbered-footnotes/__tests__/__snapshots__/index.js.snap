// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`{"gfm":false,"commonmark":false,"footnotes":true} footnote-split 1`] = `
"<p>a<sup id=\\"fnref-2\\"><a href=\\"#fn-2\\" class=\\"footnote-ref\\">2</a></sup>b<sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup>c<sup id=\\"fnref-3\\"><a href=\\"#fn-3\\" class=\\"footnote-ref\\">3</a></sup></p>
<div class=\\"footnotes\\">
<hr>
<ol>
<li id=\\"fn-1\\">
second def
<a href=\\"#fnref-1\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-2\\">
<p>first def</p>
<a href=\\"#fnref-2\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-3\\">
<p>third def</p>
<a href=\\"#fnref-3\\" class=\\"footnote-backref\\">↩</a>
</li>
</ol>
</div>"
`;

exports[`{"gfm":false,"commonmark":false,"footnotes":true} footnotes 1`] = `
"<p>This is the body with a footnote<sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup> or two<sup id=\\"fnref-2\\"><a href=\\"#fn-2\\" class=\\"footnote-ref\\">2</a></sup> or more <sup id=\\"fnref-3\\"><a href=\\"#fn-3\\" class=\\"footnote-ref\\">3</a></sup> <sup id=\\"fnref-4\\"><a href=\\"#fn-4\\" class=\\"footnote-ref\\">4</a></sup> <sup id=\\"fnref-5\\"><a href=\\"#fn-5\\" class=\\"footnote-ref\\">5</a></sup>. One again: <sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup></p>
<p>Also a reference that does not exist<sup id=\\"fnref-6\\"><a href=\\"#fn-6\\" class=\\"footnote-ref\\">6</a></sup>.</p>
<p>Second line of first paragraph is not intended.
Nor is third...</p>
<div class=\\"footnotes\\">
<hr>
<ol>
<li id=\\"fn-1\\">
<p>Footnote that ends with a list:</p>
<ul>
<li>item 1</li>
<li>item 2</li>
</ul>
<a href=\\"#fnref-1\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-2\\">
<blockquote>
<p>This footnote is a blockquote.</p>
</blockquote>
<a href=\\"#fnref-2\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-3\\">
<p>A simple oneliner.</p>
<a href=\\"#fnref-3\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-4\\">
<p>A footnote with multiple paragraphs.</p>
<p>Paragraph two.</p>
<a href=\\"#fnref-4\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-5\\">
<p>First line of first paragraph.</p>
<a href=\\"#fnref-5\\" class=\\"footnote-backref\\">↩</a>
</li>
</ol>
</div>"
`;

exports[`{"gfm":false,"commonmark":false,"footnotes":true} regression-1 1`] = `
"<p>a<sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup></p>
<p>a<sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup></p>
<p>a<sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup></p>
<p>a<sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup></p>
<p>a<sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup></p>
<p>a<sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup></p>
<div class=\\"footnotes\\">
<hr>
<ol>
<li id=\\"fn-1\\">
<p>MyNote</p>
<a href=\\"#fnref-1\\" class=\\"footnote-backref\\">↩</a>
</li>
</ol>
</div>"
`;

exports[`{"gfm":false,"commonmark":false,"footnotes":true} regression-2 1`] = `
"<p>1 <sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup>
2 <sup id=\\"fnref-2\\"><a href=\\"#fn-2\\" class=\\"footnote-ref\\">2</a></sup>
3 <sup id=\\"fnref-3\\"><a href=\\"#fn-3\\" class=\\"footnote-ref\\">3</a></sup>
4 <sup id=\\"fnref-4\\"><a href=\\"#fn-4\\" class=\\"footnote-ref\\">4</a></sup>
4 <sup id=\\"fnref-4\\"><a href=\\"#fn-4\\" class=\\"footnote-ref\\">4</a></sup>
5 <sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup></p>
<div class=\\"footnotes\\">
<hr>
<ol>
<li id=\\"fn-1\\">
alpha bravo one
<a href=\\"#fnref-1\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-2\\">
alpha bravo two
<a href=\\"#fnref-2\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-3\\">
alpha bravo third
<a href=\\"#fnref-3\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-4\\">
<p>foo</p>
<a href=\\"#fnref-4\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-1\\">
<p>alpha bravo fourth</p>
<a href=\\"#fnref-1\\" class=\\"footnote-backref\\">↩</a>
</li>
</ol>
</div>"
`;

exports[`{"gfm":false,"commonmark":true,"footnotes":true} footnote-split 1`] = `
"<p>a<sup id=\\"fnref-2\\"><a href=\\"#fn-2\\" class=\\"footnote-ref\\">2</a></sup>b<sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup>c<sup id=\\"fnref-3\\"><a href=\\"#fn-3\\" class=\\"footnote-ref\\">3</a></sup></p>
<div class=\\"footnotes\\">
<hr>
<ol>
<li id=\\"fn-1\\">
second def
<a href=\\"#fnref-1\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-2\\">
<p>first def</p>
<a href=\\"#fnref-2\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-3\\">
<p>third def</p>
<a href=\\"#fnref-3\\" class=\\"footnote-backref\\">↩</a>
</li>
</ol>
</div>"
`;

exports[`{"gfm":false,"commonmark":true,"footnotes":true} footnotes 1`] = `
"<p>This is the body with a footnote<sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup> or two<sup id=\\"fnref-2\\"><a href=\\"#fn-2\\" class=\\"footnote-ref\\">2</a></sup> or more <sup id=\\"fnref-3\\"><a href=\\"#fn-3\\" class=\\"footnote-ref\\">3</a></sup> <sup id=\\"fnref-4\\"><a href=\\"#fn-4\\" class=\\"footnote-ref\\">4</a></sup> <sup id=\\"fnref-5\\"><a href=\\"#fn-5\\" class=\\"footnote-ref\\">5</a></sup>. One again: <sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup></p>
<p>Also a reference that does not exist<sup id=\\"fnref-6\\"><a href=\\"#fn-6\\" class=\\"footnote-ref\\">6</a></sup>.</p>
<p>Second line of first paragraph is not intended.
Nor is third...</p>
<div class=\\"footnotes\\">
<hr>
<ol>
<li id=\\"fn-1\\">
<p>Footnote that ends with a list:</p>
<ul>
<li>item 1</li>
<li>item 2</li>
</ul>
<a href=\\"#fnref-1\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-2\\">
<blockquote>
<p>This footnote is a blockquote.</p>
</blockquote>
<a href=\\"#fnref-2\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-3\\">
<p>A simple oneliner.</p>
<a href=\\"#fnref-3\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-4\\">
<p>A footnote with multiple paragraphs.</p>
<p>Paragraph two.</p>
<a href=\\"#fnref-4\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-5\\">
<p>First line of first paragraph.</p>
<a href=\\"#fnref-5\\" class=\\"footnote-backref\\">↩</a>
</li>
</ol>
</div>"
`;

exports[`{"gfm":false,"commonmark":true,"footnotes":true} regression-1 1`] = `
"<p>a<sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup></p>
<p>a<sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup></p>
<p>a<sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup></p>
<p>a<sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup></p>
<p>a<sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup></p>
<p>a<sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup></p>
<div class=\\"footnotes\\">
<hr>
<ol>
<li id=\\"fn-1\\">
<p>MyNote</p>
<a href=\\"#fnref-1\\" class=\\"footnote-backref\\">↩</a>
</li>
</ol>
</div>"
`;

exports[`{"gfm":false,"commonmark":true,"footnotes":true} regression-2 1`] = `
"<p>1 <sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup>
2 <sup id=\\"fnref-2\\"><a href=\\"#fn-2\\" class=\\"footnote-ref\\">2</a></sup>
3 <sup id=\\"fnref-3\\"><a href=\\"#fn-3\\" class=\\"footnote-ref\\">3</a></sup>
4 <sup id=\\"fnref-4\\"><a href=\\"#fn-4\\" class=\\"footnote-ref\\">4</a></sup>
4 <sup id=\\"fnref-4\\"><a href=\\"#fn-4\\" class=\\"footnote-ref\\">4</a></sup>
5 <sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup></p>
<div class=\\"footnotes\\">
<hr>
<ol>
<li id=\\"fn-1\\">
alpha bravo one
<a href=\\"#fnref-1\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-2\\">
alpha bravo two
<a href=\\"#fnref-2\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-3\\">
alpha bravo third
<a href=\\"#fnref-3\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-4\\">
<p>foo</p>
<a href=\\"#fnref-4\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-1\\">
<p>alpha bravo fourth</p>
<a href=\\"#fnref-1\\" class=\\"footnote-backref\\">↩</a>
</li>
</ol>
</div>"
`;

exports[`{"gfm":true,"commonmark":false,"footnotes":true} footnote-split 1`] = `
"<p>a<sup id=\\"fnref-2\\"><a href=\\"#fn-2\\" class=\\"footnote-ref\\">2</a></sup>b<sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup>c<sup id=\\"fnref-3\\"><a href=\\"#fn-3\\" class=\\"footnote-ref\\">3</a></sup></p>
<div class=\\"footnotes\\">
<hr>
<ol>
<li id=\\"fn-1\\">
second def
<a href=\\"#fnref-1\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-2\\">
<p>first def</p>
<a href=\\"#fnref-2\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-3\\">
<p>third def</p>
<a href=\\"#fnref-3\\" class=\\"footnote-backref\\">↩</a>
</li>
</ol>
</div>"
`;

exports[`{"gfm":true,"commonmark":false,"footnotes":true} footnotes 1`] = `
"<p>This is the body with a footnote<sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup> or two<sup id=\\"fnref-2\\"><a href=\\"#fn-2\\" class=\\"footnote-ref\\">2</a></sup> or more <sup id=\\"fnref-3\\"><a href=\\"#fn-3\\" class=\\"footnote-ref\\">3</a></sup> <sup id=\\"fnref-4\\"><a href=\\"#fn-4\\" class=\\"footnote-ref\\">4</a></sup> <sup id=\\"fnref-5\\"><a href=\\"#fn-5\\" class=\\"footnote-ref\\">5</a></sup>. One again: <sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup></p>
<p>Also a reference that does not exist<sup id=\\"fnref-6\\"><a href=\\"#fn-6\\" class=\\"footnote-ref\\">6</a></sup>.</p>
<p>Second line of first paragraph is not intended.
Nor is third...</p>
<div class=\\"footnotes\\">
<hr>
<ol>
<li id=\\"fn-1\\">
<p>Footnote that ends with a list:</p>
<ul>
<li>item 1</li>
<li>item 2</li>
</ul>
<a href=\\"#fnref-1\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-2\\">
<blockquote>
<p>This footnote is a blockquote.</p>
</blockquote>
<a href=\\"#fnref-2\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-3\\">
<p>A simple oneliner.</p>
<a href=\\"#fnref-3\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-4\\">
<p>A footnote with multiple paragraphs.</p>
<p>Paragraph two.</p>
<a href=\\"#fnref-4\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-5\\">
<p>First line of first paragraph.</p>
<a href=\\"#fnref-5\\" class=\\"footnote-backref\\">↩</a>
</li>
</ol>
</div>"
`;

exports[`{"gfm":true,"commonmark":false,"footnotes":true} regression-1 1`] = `
"<p>a<sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup></p>
<p>a<sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup></p>
<p>a<sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup></p>
<p>a<sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup></p>
<p>a<sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup></p>
<p>a<sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup></p>
<div class=\\"footnotes\\">
<hr>
<ol>
<li id=\\"fn-1\\">
<p>MyNote</p>
<a href=\\"#fnref-1\\" class=\\"footnote-backref\\">↩</a>
</li>
</ol>
</div>"
`;

exports[`{"gfm":true,"commonmark":false,"footnotes":true} regression-2 1`] = `
"<p>1 <sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup>
2 <sup id=\\"fnref-2\\"><a href=\\"#fn-2\\" class=\\"footnote-ref\\">2</a></sup>
3 <sup id=\\"fnref-3\\"><a href=\\"#fn-3\\" class=\\"footnote-ref\\">3</a></sup>
4 <sup id=\\"fnref-4\\"><a href=\\"#fn-4\\" class=\\"footnote-ref\\">4</a></sup>
4 <sup id=\\"fnref-4\\"><a href=\\"#fn-4\\" class=\\"footnote-ref\\">4</a></sup>
5 <sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup></p>
<div class=\\"footnotes\\">
<hr>
<ol>
<li id=\\"fn-1\\">
alpha bravo one
<a href=\\"#fnref-1\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-2\\">
alpha bravo two
<a href=\\"#fnref-2\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-3\\">
alpha bravo third
<a href=\\"#fnref-3\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-4\\">
<p>foo</p>
<a href=\\"#fnref-4\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-1\\">
<p>alpha bravo fourth</p>
<a href=\\"#fnref-1\\" class=\\"footnote-backref\\">↩</a>
</li>
</ol>
</div>"
`;

exports[`{"gfm":true,"commonmark":true,"footnotes":true} footnote-split 1`] = `
"<p>a<sup id=\\"fnref-2\\"><a href=\\"#fn-2\\" class=\\"footnote-ref\\">2</a></sup>b<sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup>c<sup id=\\"fnref-3\\"><a href=\\"#fn-3\\" class=\\"footnote-ref\\">3</a></sup></p>
<div class=\\"footnotes\\">
<hr>
<ol>
<li id=\\"fn-1\\">
second def
<a href=\\"#fnref-1\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-2\\">
<p>first def</p>
<a href=\\"#fnref-2\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-3\\">
<p>third def</p>
<a href=\\"#fnref-3\\" class=\\"footnote-backref\\">↩</a>
</li>
</ol>
</div>"
`;

exports[`{"gfm":true,"commonmark":true,"footnotes":true} footnotes 1`] = `
"<p>This is the body with a footnote<sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup> or two<sup id=\\"fnref-2\\"><a href=\\"#fn-2\\" class=\\"footnote-ref\\">2</a></sup> or more <sup id=\\"fnref-3\\"><a href=\\"#fn-3\\" class=\\"footnote-ref\\">3</a></sup> <sup id=\\"fnref-4\\"><a href=\\"#fn-4\\" class=\\"footnote-ref\\">4</a></sup> <sup id=\\"fnref-5\\"><a href=\\"#fn-5\\" class=\\"footnote-ref\\">5</a></sup>. One again: <sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup></p>
<p>Also a reference that does not exist<sup id=\\"fnref-6\\"><a href=\\"#fn-6\\" class=\\"footnote-ref\\">6</a></sup>.</p>
<p>Second line of first paragraph is not intended.
Nor is third...</p>
<div class=\\"footnotes\\">
<hr>
<ol>
<li id=\\"fn-1\\">
<p>Footnote that ends with a list:</p>
<ul>
<li>item 1</li>
<li>item 2</li>
</ul>
<a href=\\"#fnref-1\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-2\\">
<blockquote>
<p>This footnote is a blockquote.</p>
</blockquote>
<a href=\\"#fnref-2\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-3\\">
<p>A simple oneliner.</p>
<a href=\\"#fnref-3\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-4\\">
<p>A footnote with multiple paragraphs.</p>
<p>Paragraph two.</p>
<a href=\\"#fnref-4\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-5\\">
<p>First line of first paragraph.</p>
<a href=\\"#fnref-5\\" class=\\"footnote-backref\\">↩</a>
</li>
</ol>
</div>"
`;

exports[`{"gfm":true,"commonmark":true,"footnotes":true} regression-1 1`] = `
"<p>a<sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup></p>
<p>a<sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup></p>
<p>a<sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup></p>
<p>a<sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup></p>
<p>a<sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup></p>
<p>a<sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup><sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup></p>
<div class=\\"footnotes\\">
<hr>
<ol>
<li id=\\"fn-1\\">
<p>MyNote</p>
<a href=\\"#fnref-1\\" class=\\"footnote-backref\\">↩</a>
</li>
</ol>
</div>"
`;

exports[`{"gfm":true,"commonmark":true,"footnotes":true} regression-2 1`] = `
"<p>1 <sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup>
2 <sup id=\\"fnref-2\\"><a href=\\"#fn-2\\" class=\\"footnote-ref\\">2</a></sup>
3 <sup id=\\"fnref-3\\"><a href=\\"#fn-3\\" class=\\"footnote-ref\\">3</a></sup>
4 <sup id=\\"fnref-4\\"><a href=\\"#fn-4\\" class=\\"footnote-ref\\">4</a></sup>
4 <sup id=\\"fnref-4\\"><a href=\\"#fn-4\\" class=\\"footnote-ref\\">4</a></sup>
5 <sup id=\\"fnref-1\\"><a href=\\"#fn-1\\" class=\\"footnote-ref\\">1</a></sup></p>
<div class=\\"footnotes\\">
<hr>
<ol>
<li id=\\"fn-1\\">
alpha bravo one
<a href=\\"#fnref-1\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-2\\">
alpha bravo two
<a href=\\"#fnref-2\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-3\\">
alpha bravo third
<a href=\\"#fnref-3\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-4\\">
<p>foo</p>
<a href=\\"#fnref-4\\" class=\\"footnote-backref\\">↩</a>
</li>
<li id=\\"fn-1\\">
<p>alpha bravo fourth</p>
<a href=\\"#fnref-1\\" class=\\"footnote-backref\\">↩</a>
</li>
</ol>
</div>"
`;
